File: Data Structures & Algorithms/length-of-last-word/submission-8.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        res = 0
        temp = len(s) - 1
        while temp >= 0 and s[temp] == ' ':
            temp -= 1
        while temp >= 0 and s[temp] != ' ':
            res += 1
            temp -= 1
        return res

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-8.java
class Solution {
    public int maxProfit(int[] prices) {
        int res = 0;
        int minBuy = prices[0];
        for (int price : prices) {
            int profit = price - minBuy;
            res = Math.max(res, profit);
            minBuy = Math.min(minBuy, price);
        }
        return res;
    }
}
